### PR TITLE
NATIVE-397: Separate animator logic from settings/permission logic

### PR DESCRIPTION
**Pull request by Schedulaar**

### CODE REVIEW

**Comment by maxammann** [comment]
As some of the refactoring is hard to follow I tested this in the emulator.

**Comment by klinzo** [comment]
@Schedulaar I'll merge this right?

File: src/modules/layout/components/SnackbarAnimator.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    this.setState({ displayed: null, status: 'out' })
+    this.checkForUpdate()
+  }
+
+  componentDidUpdate (prevProps: PropsType) {
+    const children = this.props.children
+    const displayed = this.state.displayed
+    if (prevProps.children !== children) {
+      if (children && displayed && getKey(children) === getKey(displayed)) {
+        this.setState({ displayed: children })
```
Comment (klinzo) [inline src/modules/layout/components/SnackbarAnimator.js]: Why/when do we need this?

Comment (Schedulaar) [reply src/modules/layout/components/SnackbarAnimator.js]: it's necessary to update the displayed node, if there was a change (for example if some props changed).

**Comment by Schedulaar** [comment]
@klinzo yup you can merge it

**Comment by klinzo** [comment]
BTW really well done @Schedulaar !
